### ISSUE TITLE
Object('JSON') related crash

### ISSUE THREAD

**Issue by filimonov** [ClickHouse/ClickHouse#40911]
repro 

```
SET allow_experimental_object_type=1;

CREATE TABLE IF NOT EXISTS json_crash
(
    json_raw String,
    json Object('JSON') MATERIALIZED json_raw,
) engine=MergeTree() ORDER BY tuple();

INSERT INTO json_crash VALUES ('{"fields": [{"value": "1234567", "field_name": "numeric_field"}, {"value": ["fancy_tag_name", "another_tag"], "field_name": "tags"}]}');
```

```
# 2022.09.01 21:20:32.991526 [ 169278 ] {} <Trace> BaseDaemon: Received signal 11
# 2022.09.01 21:20:32.991781 [ 169798 ] {} <Fatal> BaseDaemon: ########################################
# 2022.09.01 21:20:32.991894 [ 169798 ] {} <Fatal> BaseDaemon: (version 22.9.1.759 (official build), build id: 4A9014C655EC8A958D94BCFF6AD8F931EA146DDA) (from thread 169280) (query_id: 4984d0cb-b863-4123-ab5e-0d960d09143c) (query: INSERT INTO zendesk_tickets_raw VALUES) Received signal Segmentation fault (11)
# 2022.09.01 21:20:32.991942 [ 169798 ] {} <Fatal> BaseDaemon: Address: NULL pointer. Access: read. Unknown si_code.
# 2022.09.01 21:20:32.991976 [ 169798 ] {} <Fatal> BaseDaemon: Stack trace: 0xcf79b7c 0x15044392 0x150443c4 0x15044392 0x150c39c4 0x142230a3 0x1421e76d 0xcb98b5a 0xcb9604d 0xce3231e 0xcb8b997 0x13cc8687 0x13cc9001 0x13cca429 0x147ad1e4 0x15e94f01 0x15e93837 0x15e93559 0x15e92dce 0x15cbbeca 0x15cb043b 0x15caf608 0x15c547cc 0x15c4c33a 0x15c606d9 0x189ce5d3 0x189cf92d 0x18b92089 0x18b8f942 0x7efd865cdb43 0x7efd8665fa00
# 2022.09.01 21:20:36.119210 [ 169798 ] {} <Fatal> BaseDaemon: 2. DB::ColumnString::insert(DB::Field const&) @ 0xcf79b7c in /usr/bin/clickhouse
# 2022.09.01 21:20:36.771076 [ 169798 ] {} <Fatal> BaseDaemon: 3. ./build_docker/../src/Columns/ColumnArray.cpp:0: DB::ColumnArray::insert(DB::Field const&) @ 0x15044392 in /usr/bin/clickhouse
# 2022.09.01 21:20:37.358504 [ 169798 ] {} <Fatal> BaseDaemon: 4. ./build_docker/../src/Columns/ColumnArray.cpp:301: DB::ColumnArray::insert(DB::Field const&) @ 0x150443c4 in /usr/bin/clickhouse
# 2022.09.01 21:20:37.944714 [ 169798 ] {} <Fatal> BaseDaemon: 5. ./build_docker/../src/Columns/ColumnArray.cpp:0: DB::ColumnArray::insert(DB::Field const&) @ 0x15044392 in /usr/bin/clickhouse
# 2022.09.01 21:20:38.541315 [ 169798 ] {} <Fatal> BaseDaemon: 6. ./build_docker/../src/Columns/ColumnObject.cpp:0: DB::ColumnObject::Subcolumn::insert(DB::Field, DB::FieldInfo) @ 0x150c39c4 in /usr/bin/clickhouse
# 2022.09.01 21:20:38.982327 [ 169798 ] {} <Fatal> BaseDaemon: 7. ./build_docker/../src/DataTypes/Serializations/SerializationObject.cpp:82: void DB::SerializationObject<DB::JSONDataParser<DB::SimdJSONParser> >::deserializeTextImpl<DB::SerializationObject<DB::JSONDataParser<DB::SimdJSONParser> >::deserializeWholeText(DB::IColumn&, DB::ReadBuffer&, DB::FormatSettings const&) const::'lambda'(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&)>(DB::IColumn&, DB::SerializationObject<DB::JSONDataParser<DB::SimdJSONParser> >::deserializeWholeText(DB::IColumn&, DB::ReadBuffer&, DB::FormatSettings const&) const::'lambda'(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&)&&) const @ 0x142230a3 in /usr/bin/clickhouse
# 2022.09.01 21:20:39.407611 [ 169798 ] {} <Fatal> BaseDaemon: 8. ./build_docker/../src/DataTypes/Serializations/SerializationObject.cpp:104: DB::SerializationObject<DB::JSONDataParser<DB::SimdJSONParser> >::deserializeWholeText(DB::IColumn&, DB::ReadBuffer&, DB::FormatSettings const&) const @ 0x1421e76d in /usr/bin/clickhouse
# 2022.09.01 21:20:40.482433 [ 169798 ] {} <Fatal> BaseDaemon: 9. DB::ConvertImplGenericFromString<DB::ColumnString>::executeImpl(DB::IColumn const&, DB::IColumn&, DB::ISerialization const&, unsigned long, DB::PODArray<char8_t, 4096ul, Allocator<false, false>, 15ul, 16ul> const*, DB::IDataType const*) @ 0xcb98b5a in /usr/bin/clickhouse
# 2022.09.01 21:20:41.558638 [ 169798 ] {} <Fatal> BaseDaemon: 10. DB::ConvertImplGenericFromString<DB::ColumnString>::execute(std::__1::vector<DB::ColumnWithTypeAndName, std::__1::allocator<DB::ColumnWithTypeAndName> >&, std::__1::shared_ptr<DB::IDataType const> const&, DB::ColumnNullable const*, unsigned long) @ 0xcb9604d in /usr/bin/clickhouse
# 2022.09.01 21:20:42.639016 [ 169798 ] {} <Fatal> BaseDaemon: 11. COW<DB::IColumn>::immutable_ptr<DB::IColumn> std::__1::__function::__policy_invoker<COW<DB::IColumn>::immutable_ptr<DB::IColumn> (std::__1::vector<DB::ColumnWithTypeAndName, std::__1::allocator<DB::ColumnWithTypeAndName> >&, std::__1::shared_ptr<DB::IDataType const> const&, DB::ColumnNullable const*, unsigned long)>::__call_impl<std::__1::__function::__default_alloc_func<DB::FunctionCast<DB::CastInternalName>::createObjectWrapper(std::__1::shared_ptr<DB::IDataType const> const&, DB::DataTypeObject const*) const::'lambda'(std::__1::vector<DB::ColumnWithTypeAndName, std::__1::allocator<DB::ColumnWithTypeAndName> >&, std::__1::shared_ptr<DB::IDataType const> const&, DB::ColumnNullable const*, unsigned long), COW<DB::IColumn>::immutable_ptr<DB::IColumn> (std::__1::vector<DB::ColumnWithTypeAndName, std::__1::allocator<DB::ColumnWithTypeAndName> >&, std::__1::shared_ptr<DB::IDataType const> const&, DB::ColumnNullable const*, unsigned long)> >(std::__1::__function::__policy_storage const*, std::__1::vector<DB::ColumnWithTypeAndName, std::__1::allocator<DB::ColumnWithTypeAndName> >&, std::__1::shared_ptr<DB::IDataType const> const&, DB::ColumnNullable const*, unsigned long) @ 0xce3231e in /usr/bin/clickhouse
# 2022.09.01 21:20:43.720052 [ 169798 ] {} <Fatal> BaseDaemon: 12. DB::ExecutableFunctionCast::executeImpl(std::__1::vector<DB::ColumnWithTypeAndName, std::__1::allocator<DB::ColumnWithTypeAndName> > const&, std::__1::shared_ptr<DB::IDataType const> const&, unsigned long) const @ 0xcb8b997 in /usr/bin/clickhouse
# 2022.09.01 21:20:44.127635 [ 169798 ] {} <Fatal> BaseDaemon: 13. ./build_docker/../src/Functions/IFunction.cpp:0: DB::IExecutableFunction::executeWithoutLowCardinalityColumns(std::__1::vector<DB::ColumnWithTypeAndName, std::__1::allocator<DB::ColumnWithTypeAndName> > const&, std::__1::shared_ptr<DB::IDataType const> const&, unsigned long, bool) const @ 0x13cc8687 in /usr/bin/clickhouse
# 2022.09.01 21:20:44.499324 [ 169798 ] {} <Fatal> BaseDaemon: 14.1. inlined from ./build_docker/../contrib/boost/boost/smart_ptr/intrusive_ptr.hpp:115: intrusive_ptr
# 2022.09.01 21:20:44.499409 [ 169798 ] {} <Fatal> BaseDaemon: 14.2. inlined from ../contrib/boost/boost/smart_ptr/intrusive_ptr.hpp:122: boost::intrusive_ptr<DB::IColumn const>::operator=(boost::intrusive_ptr<DB::IColumn const>&&)
# 2022.09.01 21:20:44.499444 [ 169798 ] {} <Fatal> BaseDaemon: 14.3. inlined from ../src/Common/COW.h:136: COW<DB::IColumn>::immutable_ptr<DB::IColumn>::operator=(COW<DB::IColumn>::immutable_ptr<DB::IColumn>&&)
# 2022.09.01 21:20:44.499468 [ 169798 ] {} <Fatal> BaseDaemon: 14. ../src/Functions/IFunction.cpp:306: DB::IExecutableFunction::executeWithoutSparseColumns(std::__1::vector<DB::ColumnWithTypeAndName, std::__1::allocator<DB::ColumnWithTypeAndName> > const&, std::__1::shared_ptr<DB::IDataType const> const&, unsigned long, bool) const @ 0x13cc9001 in /usr/bin/clickhouse
# 2022.09.01 21:20:44.874825 [ 169798 ] {} <Fatal> BaseDaemon: 15.1. inlined from ./build_docker/../contrib/libcxx/include/vector:399: ~vector
# 2022.09.01 21:20:44.874907 [ 169798 ] {} <Fatal> BaseDaemon: 15. ../src/Functions/IFunction.cpp:373: DB::IExecutableFunction::execute(std::__1::vector<DB::ColumnWithTypeAndName, std::__1::allocator<DB::ColumnWithTypeAndName> > const&, std::__1::shared_ptr<DB::IDataType const> const&, unsigned long, bool) const @ 0x13cca429 in /usr/bin/clickhouse
# 2022.09.01 21:20:45.426619 [ 169798 ] {} <Fatal> BaseDaemon: 16.1. inlined from ./build_docker/../contrib/boost/boost/smart_ptr/intrusive_ptr.hpp:115: intrusive_ptr
# 2022.09.01 21:20:45.426693 [ 169798 ] {} <Fatal> BaseDaemon: 16.2. inlined from ../contrib/boost/boost/smart_ptr/intrusive_ptr.hpp:122: boost::intrusive_ptr<DB::IColumn const>::operator=(boost::intrusive_ptr<DB::IColumn const>&&)
# 2022.09.01 21:20:45.426742 [ 169798 ] {} <Fatal> BaseDaemon: 16.3. inlined from ../src/Common/COW.h:136: COW<DB::IColumn>::immutable_ptr<DB::IColumn>::operator=(COW<DB::IColumn>::immutable_ptr<DB::IColumn>&&)
# 2022.09.01 21:20:45.426790 [ 169798 ] {} <Fatal> BaseDaemon: 16.4. inlined from ../src/Interpreters/ExpressionActions.cpp:607: DB::executeAction(DB::ExpressionActions::Action const&, DB::(anonymous namespace)::ExecutionContext&, bool)
# 2022.09.01 21:20:45.426834 [ 169798 ] {} <Fatal> BaseDaemon: 16. ../src/Interpreters/ExpressionActions.cpp:724: DB::ExpressionActions::execute(DB::Block&, unsigned long&, bool) const @ 0x147ad1e4 in /usr/bin/clickhouse
# 2022.09.01 21:20:46.161006 [ 169798 ] {} <Fatal> BaseDaemon: 17. ./build_docker/../contrib/libcxx/include/vector:505: DB::ConvertingTransform::onConsume(DB::Chunk) @ 0x15e94f01 in /usr/bin/clickhouse
# 2022.09.01 21:20:46.911572 [ 169798 ] {} <Fatal> BaseDaemon: 18.1. inlined from ./build_docker/../contrib/libcxx/include/__memory/shared_ptr.h:702: ~shared_ptr
# 2022.09.01 21:20:46.911640 [ 169798 ] {} <Fatal> BaseDaemon: 18.2. inlined from ../src/Processors/Chunk.h:32: ~Chunk
# 2022.09.01 21:20:46.911680 [ 169798 ] {} <Fatal> BaseDaemon: 18.3. inlined from ../src/Processors/Transforms/ExceptionKeepingTransform.cpp:151: operator()
# 2022.09.01 21:20:46.911734 [ 169798 ] {} <Fatal> BaseDaemon: 18.4. inlined from ../contrib/libcxx/include/type_traits:3640: decltype(static_cast<DB::ExceptionKeepingTransform::work()::$_1&>(fp)()) std::__1::__invoke<DB::ExceptionKeepingTransform::work()::$_1&>(DB::ExceptionKeepingTransform::work()::$_1&)
# 2022.09.01 21:20:46.911780 [ 169798 ] {} <Fatal> BaseDaemon: 18.5. inlined from ../contrib/libcxx/include/__functional/invoke.h:61: void std::__1::__invoke_void_return_wrapper<void, true>::__call<DB::ExceptionKeepingTransform::work()::$_1&>(DB::ExceptionKeepingTransform::work()::$_1&)
# 2022.09.01 21:20:46.911825 [ 169798 ] {} <Fatal> BaseDaemon: 18.6. inlined from ../contrib/libcxx/include/__functional/function.h:230: std::__1::__function::__default_alloc_func<DB::ExceptionKeepingTransform::work()::$_1, void ()>::operator()()
# 2022.09.01 21:20:46.911873 [ 169798 ] {} <Fatal> BaseDaemon: 18. ../contrib/libcxx/include/__functional/function.h:711: void std::__1::__function::__policy_invoker<void ()>::__call_impl<std::__1::__function::__default_alloc_func<DB::ExceptionKeepingTransform::work()::$_1, void ()> >(std::__1::__function::__policy_storage const*) @ 0x15e93837 in /usr/bin/clickhouse
# 2022.09.01 21:20:47.641359 [ 169798 ] {} <Fatal> BaseDaemon: 19. ./build_docker/../src/Processors/Transforms/ExceptionKeepingTransform.cpp:103: DB::runStep(std::__1::function<void ()>, DB::ThreadStatus*, std::__1::atomic<unsigned long>*) @ 0x15e93559 in /usr/bin/clickhouse
# 2022.09.01 21:20:48.360191 [ 169798 ] {} <Fatal> BaseDaemon: 20.1. inlined from ./build_docker/../contrib/libcxx/include/__functional/function.h:813: ~__policy_func
# 2022.09.01 21:20:48.360257 [ 169798 ] {} <Fatal> BaseDaemon: 20.2. inlined from ../contrib/libcxx/include/__functional/function.h:1171: ~function
# 2022.09.01 21:20:48.360284 [ 169798 ] {} <Fatal> BaseDaemon: 20. ../src/Processors/Transforms/ExceptionKeepingTransform.cpp:151: DB::ExceptionKeepingTransform::work() @ 0x15e92dce in /usr/bin/clickhouse
# 2022.09.01 21:20:49.066072 [ 169798 ] {} <Fatal> BaseDaemon: 21.1. inlined from ./build_docker/../src/Processors/Executors/ExecutionThreadContext.cpp:0: DB::executeJob(DB::ExecutingGraph::Node*, DB::ReadProgressCallback*)
# 2022.09.01 21:20:49.066297 [ 169798 ] {} <Fatal> BaseDaemon: 21. ../src/Processors/Executors/ExecutionThreadContext.cpp:92: DB::ExecutionThreadContext::executeTask() @ 0x15cbbeca in /usr/bin/clickhouse
# 2022.09.01 21:20:49.782781 [ 169798 ] {} <Fatal> BaseDaemon: 22. ./build_docker/../src/Processors/Executors/PipelineExecutor.cpp:228: DB::PipelineExecutor::executeStepImpl(unsigned long, std::__1::atomic<bool>*) @ 0x15cb043b in /usr/bin/clickhouse
# 2022.09.01 21:20:50.486091 [ 169798 ] {} <Fatal> BaseDaemon: 23.1. inlined from ./build_docker/../contrib/libcxx/include/atomic:952: bool std::__1::__cxx_atomic_load<bool>(std::__1::__cxx_atomic_base_impl<bool> const*, std::__1::memory_order)
# 2022.09.01 21:20:50.486163 [ 169798 ] {} <Fatal> BaseDaemon: 23.2. inlined from ../contrib/libcxx/include/atomic:1582: std::__1::__atomic_base<bool, false>::load(std::__1::memory_order) const
# 2022.09.01 21:20:50.486207 [ 169798 ] {} <Fatal> BaseDaemon: 23.3. inlined from ../contrib/libcxx/include/atomic:1586: std::__1::__atomic_base<bool, false>::operator bool() const
# 2022.09.01 21:20:50.486253 [ 169798 ] {} <Fatal> BaseDaemon: 23.4. inlined from ../src/Processors/Executors/ExecutorTasks.h:53: DB::ExecutorTasks::isFinished() const
# 2022.09.01 21:20:50.486305 [ 169798 ] {} <Fatal> BaseDaemon: 23. ../src/Processors/Executors/PipelineExecutor.cpp:127: DB::PipelineExecutor::executeStep(std::__1::atomic<bool>*) @ 0x15caf608 in /usr/bin/clickhouse
# 2022.09.01 21:20:51.236002 [ 169798 ] {} <Fatal> BaseDaemon: 24. ./build_docker/../src/Processors/Executors/PushingPipelineExecutor.cpp:0: DB::TCPHandler::processInsertQuery() @ 0x15c547cc in /usr/bin/clickhouse
# 2022.09.01 21:20:51.940043 [ 169798 ] {} <Fatal> BaseDaemon: 25. ./build_docker/../src/Server/TCPHandler.cpp:0: DB::TCPHandler::runImpl() @ 0x15c4c33a in /usr/bin/clickhouse
# 2022.09.01 21:20:52.661319 [ 169798 ] {} <Fatal> BaseDaemon: 26. ./build_docker/../src/Server/TCPHandler.cpp:1869: DB::TCPHandler::run() @ 0x15c606d9 in /usr/bin/clickhouse
# Segmentation fault (core dumped)
```

All versions which have JSON are affected.

**Comment by filimonov** [comment]
p.s. if you try to insert those as 2 separate rows - it reports the error:

```
INSERT INTO json_crash VALUES ('{"fields": [{"value": ["fancy_tag_name", "another_tag"], "field_name": "tags"}]}');
INSERT INTO json_crash VALUES ('{"fields": [{"value": "1234567", "field_name": "numeric_field"}]}');

Received exception from server (version 22.9.1):
Code: 53. DB::Exception: Received from localhost:9000. DB::Exception: Uncompatible types of subcolumn 'fields.value': Array(String) and Array(Array(String)). (TYPE_MISMATCH)
```

**Comment by canhld94** [comment]
The core has been solved in #40483 by normalizing dimension of all value in same path. But from your 2nd example seem there'll be an inconsistency behavior when inserting (all in one vs separated rows)..

**Comment by filimonov** [comment]
I confirm that after #40483 server does not crash anymore, and data properly inserted (String and Array(String) were 'upscaled' to Array). 

But it still complains when 2 rows were inserted one after another.
